Draft-11 Release notes (#963)
moq-wg/moq-transport: draft-ietf-moq-transport.md

- Mike English
- Mo Zanaty
- Will Law

--- back

# Change Log

> RFC Editor's Note: Please remove this section prior to publication of a final version of this document.

Reviewer: ```suggestion
RFC Editor's Note: Please remove this section prior to publication of a final version of this document.

Issue and pull request numbers are listed with a leading octothorp.
```

## Since draft-ietf-moq-transport-10

* Added Common Structure definitions - Location, Key-Value-Pair and Reason
  Phrase

> * Length limits on all variable length fields, including Track Namespace and

Reviewer: ```suggestion
* Limit lengths of all variable length fields, including Track Namespace and
```

  Names
* Control Message length is now 16 bits instead of variable length

> * Subscribe ID became Request ID, added to most control messages, used now to
>   correlate OK/ERROR responses for ANNOUNCE, SUBSCRIBE_ANNOUNCES, TRACK_STATUS

Reviewer: ```suggestion
* Subscribe ID became Request ID, and was added to most control messages. Request ID
  is used to correlate OK/ERROR responses for ANNOUNCE, SUBSCRIBE_ANNOUNCES,
  and TRACK_STATUS.  Like Subscribe ID, Request IDs are flow controlled.
```

* Explain rules for caching in more detail

> * Changed SETUP parameter format for even number parameters to match Extension
>   format

Reviewer: ```suggestion
* Changed the SETUP parameter format for even number parameters to match the
  Object Header Extension format
```

* Rotated SETUP code points
* Added Parameters to TRACK_STATUS and TRACK_STATUS_REQUEST
* Clarified how subscribe filters work
* Added Next Group Filter to SUBSCRIBE
* Added Forward flag to SUBSCRIBE
* Renamed FETCH_OK field to End and clarified how to set it
* Added Absolute Joining Fetch
* Clarified No Error vs Invalid Range FETCH_ERROR cases
* Use bits in SUBGROUP_HEADER and DATAGRAM* types to compress subgroup ID and
  extensions
* Coalesced END_OF_GROUP and END_OF_TRACK_AND_GROUP status
* Objects that Do Not Exist cannot have extensions when sent on the wire
* Specified error codes for resetting data streams

> * Defined an extension for communicating a known Group ID gap

Reviewer: ```suggestion
* Defined an Object Header Extension for communicating a known Group ID gap
```

> * Replaced AUTH INFO with AUTH TOKEN, which has more structure, compression and
>   additional Auth related error codes

Reviewer: ```suggestion
* Replaced AUTHORIZATION_INFO with AUTHORIZATION_TOKEN, which has more structure, 
  compression and additional Auth related error codes (#760)
```

- Mike English
- Mo Zanaty
- Will Law

--- back

# Change Log

RFC Editor's Note: Please remove this section prior to publication of a final version of this document.

Issue and pull request numbers are listed with a leading octothorp.

## Since draft-ietf-moq-transport-10

* Added Common Structure definitions - Location, Key-Value-Pair and Reason
  Phrase

> * Limit lengths of all variable length fields, including Track Namespace and
>   Names

Reviewer: ```suggestion
* Limit lengths of all variable length fields, including Track Namespace and Name
```

* Control Message length is now 16 bits instead of variable length
* Subscribe ID became Request ID, and was added to most control messages. Request ID
  is used to correlate OK/ERROR responses for ANNOUNCE, SUBSCRIBE_ANNOUNCES,
  and TRACK_STATUS.  Like Subscribe ID, Request IDs are flow controlled.
* Explain rules for caching in more detail
* Changed the SETUP parameter format for even number parameters to match the
  Object Header Extension format
* Rotated SETUP code points
* Added Parameters to TRACK_STATUS and TRACK_STATUS_REQUEST
* Clarified how subscribe filters work
* Added Next Group Filter to SUBSCRIBE
* Added Forward flag to SUBSCRIBE
* Renamed FETCH_OK field to End and clarified how to set it
* Added Absolute Joining Fetch
* Clarified No Error vs Invalid Range FETCH_ERROR cases
* Use bits in SUBGROUP_HEADER and DATAGRAM* types to compress subgroup ID and
  extensions
* Coalesced END_OF_GROUP and END_OF_TRACK_AND_GROUP status
* Objects that Do Not Exist cannot have extensions when sent on the wire
* Specified error codes for resetting data streams
* Defined an Object Header Extension for communicating a known Group ID gap

> * Replaced AUTHORIZATION_INFO with AUTHORIZATION_TOKEN, which has more structure, 
>   compression and additional Auth related error codes (#760)

Reviewer: ```suggestion
* Replaced AUTHORIZATION_INFO with AUTHORIZATION_TOKEN, which has more structure,
  compression, and additional Auth related error codes (#760)
```
